definitions/npm/ramda_v0.x.x/flow_v0.39.x-/test_ramda_v0.x.x_functions.js

```diff
@@ -284,3 +284,91 @@ const needs3_error2: string => string = uncurryN(3, needs3)('', '', '')
 const needs3_error3: string => string => string = uncurryN(3, needs3)('', '', '')
 
 const needs3_no_error: string = uncurryN(3, needs3)('', '', '')
+
+// --- PipeP ---
+const pAdd: (number) => (number) => Promise<number> = (a) => (b) => Promise.resolve(a + b)
+const pAppend: (string) => (string) => Promise<string> = (a) => (b) => Promise.resolve(a + b)
+const pToString: (number) => Promise<string> = (n) => Promise.resolve(String(n))
+const pFn1: (number) => Promise<number> =  pipeP(pAdd(2))
+const pFoo1: Promise<number> = pFn1(2)
+const pFn2: (number) => Promise<string> =  pipeP(pAdd(2), pToString)
+const pFoo2: Promise<string> = pFn2(2)
+const pFn3: (number) => Promise<string> =  pipeP(pAdd(2), pToString, pAppend('A'))
+const pFoo3: Promise<string> = pFn3(2)
+const pFn4: (number) => Promise<string> =  pipeP(pAdd(2), pToString, pAppend('A'), pAppend('B'))
+const pFoo4: Promise<string> = pFn4(2)
+const pFn5: (number) => Promise<string> =  pipeP(pAdd(2), pToString, pAppend('A'), pAppend('B'), pAppend('C'))
+const pFoo5: Promise<string> = pFn5(2)
+const pFn6: (number) => Promise<string> =  pipeP(pAdd(2), pToString, pAppend('A'), pAppend('B'), pAppend('C'), pAppend('D'))
+const pFoo6: Promise<string> = pFn6(2)
+
+//$ExpectError
+const pipeP_error1: number => Promise<number> = _.pipeP(_.add(2))
+//$ExpectError
+const pipeP_error2: number => Promise<string> = _.pipeP(pAdd(2), String)
+//$ExpectError
+const pipeP_error3: (number) => Promise<number> =  pipeP(pAdd(2), pToString, pAdd(2))
+//$ExpectError
+const pipeP_error4: (number) => Promise<string> =  pipeP(pAdd(2), pToString, pAppend('A'), pAdd(2))
+//$ExpectError
+const pipeP_error5: (number) => Promise<string> =  pipeP(pAdd(2), pToString, pAppend('A'), pAppend('B'), pAdd(2))
+//$ExpectError
+const pipeP_error6: (number) => Promise<string> =  pipeP(pAdd(2), pToString, pAppend('A'), pAppend('B'), pAppend('C'), pAdd(2))
+
+
+// -------------
+
+// --- TyrCatch ---
+const tryFn: (boolean) => string =
+  (a) => String(a)
+
+const tryFn_2: (boolean) => boolean =
+  (a) => a
+
+const catchFn: (Error, boolean) => string =
+  (e, a) => e.message
+
+const catchFn_2: (Error, boolean) => boolean =
+  (e, a) => a
+
+const tc1: string = tryCatch(tryFn, catchFn, true);
+const tc2: boolean => string = tryCatch(tryFn, catchFn);
+const tc3: string = tc2(true);
+const tc4: string = tryCatch(tryFn, catchFn)(true);
```

Comment:
This case, tc4, *seems* like it's the same as tc2 + tc3.  Is there a reason to have both?